components/webgpu/swapchain.rs

```diff
@@ -39,6 +40,40 @@ impl MallocSizeOf for WebGPUContextId {
 
 pub type WGPUImageMap = Arc<Mutex<HashMap<WebGPUContextId, PresentationData>>>;
 
+struct GPUPresentationBuffer {
+    global: Arc<Global>,
+    buffer_id: id::BufferId,
+    data: NonNull<u8>,
+    size: usize,
+}
+
+// This is safe because [`GPUPresentationBuffer`] is always wrapped in Arc Mutex of [`WGPUImageMap`]
+unsafe impl Send for GPUPresentationBuffer {}
```

Comment:
Actually this comment is wrong, we do need to impl Send/Sync (due to pointer) because Arc/Mutex wont give them to us, but the reason this is safe is because we do not impl clone or copy, so there is only one consumer of the pointer possible.

Reply:
I think the question is rather whether there can be multiple concurrent calls to `GPUPresentationBuffer::new` for the same id, since those will have pointers to the same data. So moving the `new` call to inside the critical section, and keeping the `drop` inside it as well, is probably a good idea. I think you also want to check that the id is available before calling `new`.

Reply:
> I think you also want to check that the id is available before calling new.

Id must actually be in `queued_buffer_ids`, but this "state" is more then just buffer being mapped. I think I will just keep it as is and update comment.